src/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv/Internal/LibuvConnection.cs

```diff
@@ -80,9 +80,17 @@ public LibuvConnection()
                 }
                 finally
                 {
-                    // Ensure the socket is disposed prior to completing in the input writer.
-                    _socket.Dispose();
+                    // Make sure it isn't possible for a paused read to resume reading after calling uv_close
+                    // on the stream handle
+                    Input.CancelPendingFlush();
```

Comment:
Is there a Pipe test that verifies that the writer in fact observes cancellation when:

1. The writer continuation is scheduled after a call to IPipeReader.Advance.
2. Next CancelPendingFlush is called.
3. Lastly the _writerScheduler actually runs the continuation.

It looks like this is already the case, but it's important this never regresses.